src/app/routes/users/UserRoutes.tsx

```diff
@@ -1,12 +1,11 @@
 import React from "react";
 import { Routes, Route } from "react-router-dom";
-import SignIn from "../../pages/signin/SignIn";
-import SignUp from "../../pages/signup/SignUp";
+
+import Profile from "../../pages/users/Profile";
 function UserRoutes() {
   return (
     <Routes>
-      <Route path="/signin" element={<SignIn />} />
-      <Route path="/signup" element={<SignUp />} />
+      <Route path=":uid" element={<Profile />} />

```

Comment:
If we have a username as a primary key I am not sure we need a uid for each user; the username can be their "uid"